Spec Guidelines for new contributors (#2542)
mozilla/pontoon: specs/0112-new-contributor-guidelines.md

- Feature Name: Guidelines for new contributors
- Created: 2022-05-31
- Associated Issue: #2305

# Summary

Provide new contributors with guidelines when making their first contribution.

# Motivation

It has been suggested on various occasions and from various sources, including at the localizer workshops, that the onboarding experience for new contributors needs improvements.

Pontoon homepage and tour lead users towards making their first contribution, but they don't get much guidance since that point nor are they connected with their team. Locale Managers also don't get any direct information when new contributors join their teams.

# Feature explanation

> After the user makes her first suggestion in Pontoon, we send an in-app notification with the following goal:

Reviewer: "her first suggestion" > "their"?

- Welcome user to Pontoon.
- Clarify what will happen with the suggestion.
- Link to Pontoon docs.
- Explain how to get in touch with locale managers.

> - If set, include links to team Info tab and locale style guide.

Reviewer: Good point. Agree here as well

> - If set, include links to team Info tab and locale style guide.

Author: Could you please clarify that do you mean with the "template" info here?

> After the user makes her first suggestion in a particular locale, we send an in-app notification with just the locale-specific bits of the above defined message. We also send an in-app notification to locale managers, informing them about the new user.

Reviewer: "her first suggestion" = "their" here as well?

- Feature Name: Team guidelines for new contributors
- Created: 2022-05-31
- Associated Issue: #2305

# Summary

Provide new contributors with guidelines before making their first contribution to a team.

# Motivation

It has been suggested on various occasions and from various sources, including at the localizer workshops, that the onboarding experience for new contributors needs improvements.

Pontoon homepage and tour lead users towards making their first contribution, but don't provide any team-specific information.

On top of that, new users don't get connected with their team, and team managers don't get any explicit information about new contributors joining their teams.

# Feature explanation

## Translate view tooltip

Before users submit their first suggestion to a particular team, a tooltip appears near the text editor, with the following content:

> It looks like you haven't contributed to this locale yet. Be sure to:

Reviewer 2: I'd replace this with

```
It looks like you haven't contributed to this locale yet. 

Make sure to check the <a href="/locale-code/info">team information</a> page before starting, as it might contain important information and language resources.

If you want to contact the team managers, search for “manager” in the list of <a href="/locale-code/contributors"> active contributors for this locale</a>.
```

- Feature Name: Team guidelines for new contributors
- Created: 2022-05-31
- Associated Issue: #2305

# Summary

Provide new contributors with guidelines before making their first contribution to a team.

# Motivation

It has been suggested on various occasions and from various sources, including at the localizer workshops, that the onboarding experience for new contributors needs improvements.

Pontoon homepage and tour lead users towards making their first contribution, but don't provide any team-specific information.

On top of that, new users don't get connected with their team, and team managers don't get any explicit information about new contributors joining their teams.

# Feature explanation

## Translate view tooltip

Before users submit their first suggestion to a particular team, a tooltip appears near the text editor, with the following content:

It looks like you haven't contributed to this locale yet.

Make sure to check the <a href="/locale-code/info">team information</a> page before starting, as it might contain important information and language resources.

> If you want to contact the team managers, check out the <a href="/locale-code/contributors">team contributors</a> page.

Reviewer 3: Would we want to include examples of why someone might want to contact a team manager? If for the purpose of driving engagement/retention of a new potential volunteer, it might be better to phrase this as a call to action. 
Perhaps something like:

`Reach out to your language's team managers from the <a href="/locale-code/contributors">team contributors</a> page if you have questions or want to learn more about contributing.`

## Notification for team managers

After users submit their first suggestion to a particular team, an in-app notification is sent to team managers, informing them about the new user. Example text:

> John Smith has made their first contribution to Locale X (locale-code). Please welcome them to the team!

Reviewer 2: ```suggestion
John Smith has made their first contribution to Locale X (locale-code). Please welcome them to the team, and make sure to review their suggestions.
```

Maybe more explicit. Too much?

# Prerequisites

> 1. Before launch, we give locale managers a heads-up of how the Team Info tab content will be used in the new feature and offer them an opportunity for any updates.

Reviewer 2: ```suggestion
1. Before launch, we give locale managers a heads-up of how the Team Info tab content will be used in the new feature and offer them an opportunity to update it.
```

2. [Group contributors by role](https://github.com/mozilla/pontoon/issues/2157) in the Team Contributors page.

# Mockup

Please note that the text in the bubble might be out of date. The canonical version is specified above.

> ![](0112/mockup.png)

Reviewer 4: The mockup should indicate that this is a Suggestion, not a Translation, assuming to an existing locale.

Author: Nice catch!

- Feature Name: Team guidelines for new contributors
- Created: 2022-05-31
- Associated Issue: #2305

# Summary

Provide new contributors with guidelines before making their first contribution to a team.

# Motivation

It has been suggested on various occasions and from various sources, including at the localizer workshops, that the onboarding experience for new contributors needs improvements.

Pontoon homepage and tour lead users towards making their first contribution, but don't provide any team-specific information.

On top of that, new users don't get connected with their team, and team managers don't get any explicit information about new contributors joining their teams.

# Feature explanation

## Translate view tooltip

> Before users submits their first suggestion to any team, a tooltip appears near the text editor, with the following content:

Author: We need a tangible value proposition for that, because it's not trivial to implement editing ability. And even if we had it, judging by the Info tab edits we can expect the majority of locales will not use it.

Why can't we put the same information into the resources we link to, like GitHub does (Info tab, Style Guides)?

> Before users submits their first suggestion to any team, a tooltip appears near the text editor, with the following content:

Reviewer 2: Because, for one, I would like to display that message in my own language, and be able to try different variants if it doesn't work, or add/remove information.

GitHub doesn't have anything localized, and it doesn't deal with localization as a primary function, so I don't think we should use it as a reference besides UI/UX ideas.

If we can't have it localizable, then we should keep this info to a minimum, and just link to the team info panel. 
* We link to style guides, but very few locales have one, and it's not necessarily maintained (which is a different conversation…).
* We link to contributors, but I don't believe that's helpful for newcomers, as there's no way to only display the managers for that locale (lot of people listed, including admins).

> Before users submits their first suggestion to any team, a tooltip appears near the text editor, with the following content:

Author: That sounds more like we should finally start [localizing](https://github.com/mozilla/pontoon/issues?q=is%3Aopen+is%3Aissue+label%3Al10n) Pontoon or at least the translate view, where most of the i18n work has already been done.

I agree about the issue with the Contributors tab. To include that link, we should fix #2157 first.

I would keep the style guides, though, but only where applicable, obviously. The concerns you raise are valid for the Info tab, too. And, we already link to them from the locale dashboard.

> Before users submits their first suggestion to any team, a tooltip appears near the text editor, with the following content:

Reviewer 2: > I would keep the style guides, though, but only where applicable, obviously. The concerns you raise are valid for the Info tab, too. And, we already link to them from the locale dashboard.

Good point on the team info. Maybe we should add information in English for all locales if it's empty 🤔 

I really think that we shouldn't link to the style guides, because IMO that was done in the wrong way and it might even contain misleading information :-(

> Before users submits their first suggestion to any team, a tooltip appears near the text editor, with the following content:

Author: Should we remove the style guides links from the DB (i.e. dashboards), too?
